File: web/apps/console/src/types/media.ts
// 媒体文件相关类型定义（与后端 Go DTO 完全对应）

// 媒体项（对应后端 MediaItemDto）
export interface MediaItem {
	id: number;
	originalName: string;
	filename: string;
	fileURL: string;
	filetype: string; // image, video, document, audio
	filesize: number;
	extension: string;
	uploadedAt: string;
}

// 上传响应（对应后端 UploadResponseDto）
export interface UploadResponse {
	id: number;
	originalName: string;
	filename: string;
	fileURL: string;
	filetype: string;
	filesize: number;
	extension: string;
	uploadedAt: string;
}

// 媒体查询参数（对应后端 MediaQueryDto）
export interface MediaQueryParams {
	filetype?: "image" | "video" | "document" | "audio";
	page?: number;
	pageSize?: number;
	uploadedBy?: number;
}

// 媒体列表响应（对应后端 MediaListResponseDto）
export interface MediaListResponse {
	items: MediaItem[];
	total: number;
	page: number;
	size: number;
}

// 批量上传响应（对应后端 BatchUploadResponseDto）
export interface BatchUploadResponse {
	successCount: number;
	errorCount: number;
	results: UploadResponse[];
	errors?: string[];
}

File: server/internal/dto/media.go
package dto

type UploadResponseDto struct {
	ID           uint   `json:"id"`
	OriginalName string `json:"originalName"`
	Filename     string `json:"filename"`
	FileURL      string `json:"fileURL"`
	Filetype     string `json:"filetype"`
	Filesize     int64  `json:"filesize"`
	Extension    string `json:"extension"`
	UploadedAt   string `json:"uploadedAt"`
}

type MediaQueryDto struct {
	Filetype   string `form:"filetype" binding:"omitempty,oneof=image video document audio"`
	Page       int    `form:"page" binding:"omitempty,min=1"`
	PageSize   int    `form:"pageSize" binding:"omitempty,min=1,max=100"`
	UploadedBy uint   `form:"uploadedBy" binding:"omitempty,min=1"`
}

func (q *MediaQueryDto) Normalize() {
	if q.Page <= 0 {
		q.Page = 1
	}
	if q.PageSize <= 0 {
		q.PageSize = 20
	}
}

type MediaItemDto struct {
	ID           uint   `json:"id"`
	OriginalName string `json:"originalName"`
	Filename     string `json:"filename"`
	FileURL      string `json:"fileURL"`
	Filetype     string `json:"filetype"`
	Filesize     int64  `json:"filesize"`
	Extension    string `json:"extension"`
	UploadedAt   string `json:"uploadedAt"`
}

type MediaListResponseDto struct {
	Items []MediaItemDto `json:"items"`
	Total int            `json:"total"`
	Page  int            `json:"page"`
	Size  int            `json:"size"`
}

type BatchUploadResponseDto struct {
	SuccessCount int                 `json:"successCount"`
	ErrorCount   int                 `json:"errorCount"`
	Results      []UploadResponseDto `json:"results"`
	Errors       []string            `json:"errors,omitempty"`
}
